src/ResultSet.jl

```diff
@@ -96,26 +107,18 @@ function Base.show(io::IO, rs::ResultSet)
     print(io, "Columns ")
     for i in 1:m
         i > 1 && print(io, ", ")
-        print(io, i, ":", rs.names[i])
+        print(io, i, ":", names(rs)[i])
     end
-    m < length(rs.names) && print(io, " …")
+    m < length(names(rs)) && print(io, " …")
     println(io)
     for i in 1:n
         print(io, i, ": ")
         for j in 1:m
             j > 1 && print(io, ", ")
-            print(io, rs.columns[j][i])
+            print(io, columns(rs)[j][i])
         end
         println(io)
     end
     n < size(rs, 1) && println(io, "⋮")
 end
 
-# IteratableTables
```

Comment:
Note if you'd still like to keep explicit IterableTables compatibility, you can use some of the convenience functions provided by Tables. DataFrames, for example, defines:
```julia
IteratorInterfaceExtensions.getiterator(df::AbstractDataFrame) = Tables.datavaluerows(columntable(df))
IteratorInterfaceExtensions.isiterable(x::AbstractDataFrame) = true
TableTraits.isiterabletable(x::AbstractDataFrame) = true
```
You'd have to add `IteratorInterfaceExtensions` and `TableTraits` as explicit dependencies, but just replace `AbstractDataFrame` with `ResultSet` and it should work.